src/scanner/index.js

```diff
@@ -32,8 +32,15 @@ export async function scan(hostname, options) {
     throw new Error("The site seems to be down.");
   }
 
-  if (r.responses.auto.status < 200 || r.responses.auto.status >= 300) {
-    throw new Error("Site did not respond with a 2xx HTTP status code.");
+  // We allow 401 and 403 status codes
+  if (
+    r.responses.auto.status < 200 ||
+    (r.responses.auto.status >= 400 &&
+      ![401, 403].includes(r.responses.auto.status))
+  ) {
+    throw new Error(
+      `Site did respond with an unexpected HTTP status code ${r.responses.auto.status}.`
+    );
```

Comment:
Nit: The change seems to also permit 3xx.

PS: I'd extract a `const { status } = r.responses.auto;` to make the condition more readable.

```suggestion
  // We allow 3xx, 401 and 403 status codes.
  if (
    r.responses.auto.status < 200 ||
    (r.responses.auto.status >= 400 &&
      ![401, 403].includes(r.responses.auto.status))
  ) {
    throw new Error(
      `Site did respond with an unexpected HTTP status code ${r.responses.auto.status}.`
    );
```

Reply:
Actually, we allow 2xx, 3xx, 401 and 403. Agree on clarifying the code, applied your suggestions.

Reply:
On the scanner unit tests, these are very hard to do without an actual server that is returning defined status codes _and_ is resolvable through a proper dns entry _and_ has valid TLS certificates. It is also stretching the concept of a unit test I think. Using a service like httpbin also does not work because observatory (rightfully) does not allow paths or parameters to be passed into the scanner.

I agree that this is something worth looking into, but not in this context. The original observatory code also did not do this, probably for similar reasons.

There are several paths to solve it: 
1. Spin up actual public servers with TLS certificates and public dns names for each test scenario. This is the cleanest approach, but would incur infrastructure costs and labour.
2. Spinning up local servers and manipulate the entry for `127.0.0.1` in the system's `resolv.conf` to attach host names other than `localhost` to the local address (not really a fan of this).
3. Injecting a custom resolver into the `HttpAgent` in the axios instances. This would poison the code with test-only parts, which I am also not a fan of.
4. Something involving a proxy server, probably needing similar test-only branches in the code.

All local hackery (2, 3, 4) would mean that TLS requests would be impossible to do correctly, a major downside.